src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useReducer } from 'react';
+import PropTypes from 'prop-types';
+import StarIcon from '~components/dev-hub/icons/star-icon';
+import styled from '@emotion/styled';
+import { size, fontSize, screenSize } from './theme';
+
+const StyledContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+
+    @media ${screenSize.mediumAndUp} {
+        flex-direction: row;
+    }
+`;
+
+const StyledStar = styled(StarIcon)`
+    width: ${size.mediumLarge};
+    height: ${size.mediumLarge};
+    margin-right: ${size.mediumLarge};
+    box-sizing: content-box;
+
+    @media ${screenSize.mediumAndUp} {
+        padding: 4px;
+        width: auto;
+        height: auto;
+        margin-right: ${({ isActive }) => (isActive ? '0' : size.xsmall)};
+
+        &:hover {
+            cursor: ${({ isClicked }) => (isClicked ? 'default' : 'pointer')};
+        }
+    }
+`;
+
+const StyledStarsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    @media ${screenSize.mediumAndUp} {
+        margin-left: ${size.large};
+    }
+`;
+
+const StyledTitle = styled.span`
+    font-family: 'Fira Mono', monospace;
+    font-weight: bold;
+    font-size: ${fontSize.small};
+`;
+
+const initialState = {
+    first: false,
+    second: false,
+    third: false,
+    fourth: false,
+    fifth: false,
+    isClicked: false,
+};
+
+const reducer = (state, action) => {
+    let nextState = { ...initialState };
+    switch (action) {
+        case 'fifth':
+            nextState = { ...nextState, fifth: true };
+        case 'fourth':
+            nextState = { ...nextState, fourth: true };
+        case 'third':
+            nextState = { ...nextState, third: true };
+        case 'second':
+            nextState = { ...nextState, second: true };
+        case 'first':
+            nextState = { ...nextState, first: true };
+            break;
+        case 'clicked':
+            nextState = { ...state, isClicked: true };
+            break;
+        case 'clear':
```

Comment:
Could we use something similar to an `enum` for these so we can refer to them more easily? One pattern that may be helpful is using an object for keys/values and referring to keys:

```
const STAR_ACTIONS = {
  FIRST: 'first',
  SECOND: 'second',
  ...
  CLEAR: 'clear'
}
```